### ISSUE TITLE
Running ts-watch on Docker throws "Error: listen EADDRINUSE: address already in use" 

### ISSUE THREAD

**Issue by bahatron** [gilamran/tsc-watch#129]
Running ts-watch from my file system works like a charm. But Running ts-watch inside Docker will throw EADDRINUSE. Any way to fix this?

**Comment by gilamran** [comment]
We are talking about `tsc-watch` and not `ts-watch` right?

**Comment by bahatron** [comment]
yep sorry, 

here's the same project I'm using: https://github.com/bahatron/url-shortener-node

**Comment by gilamran** [comment]
can you try using `tsc --watch` (Without the `--onSuccess` that `tsc-watch` has) just to see where's the issue

**Comment by bahatron** [comment]
it works for me if I do something like `tsc --watch & pm2-runtime start` or `tsc --watch & nodemon`

**Comment by gilamran** [comment]
Maybe you can provide a minimal repo. to reproduce?

**Comment by bahatron** [comment]
Sure

Please visit: https://github.com/bahatron/tsc-watch-docker-example

**Comment by gilamran** [comment]
Thanks for the example repo.

I've found 2 things in your example repo.
* In your `tsconfig.json` you did not specify the `files` so typescript (tsc-watch) was watching all your files, including the dist folder, causing it to detect a file was changed in the `dist` folder. which caused it to recompile again...
* It seems like `tsc-watch` was unable to terminate the previous `npm start` for some reason.  your `--onSuccess` was to run the npm script `npm start`. I changed it to run `node dist/index.js` directly. and it was able to terminate it.

Hopes this helps
